crates/tx5/src/backend/mem.rs

```diff
@@ -0,0 +1,333 @@
+
+pub fn default_config() -> serde_json::Value {
+    serde_json::json!({})
+}
+
+pub async fn connect(
+    _config: &Arc<Config>,
+    _url: &str,
+    _listener: bool,
+) -> Result<(DynBackEp, DynBackEpRecvCon)> {
+    let (ep, ep_recv) = STAT.listen();
+    let ep: DynBackEp = ep;
+    let ep_recv: DynBackEpRecvCon = Box::new(ep_recv);
+
+    Ok((ep, ep_recv))
+}
+
+struct ConRecvData(tokio::sync::mpsc::UnboundedReceiver<Vec<u8>>);
+
+impl BackConRecvData for ConRecvData {
+    fn recv(&mut self) -> BoxFuture<'_, Option<Vec<u8>>> {
+        Box::pin(async { self.0.recv().await })
+    }
+}
+
+struct Con {
+    pub_key: PubKey,
+    send: tokio::sync::mpsc::UnboundedSender<Vec<u8>>,
+}
+
+impl Con {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(pub_key: PubKey) -> (DynBackCon, DynBackConRecvData) {
+        let (send, recv) = tokio::sync::mpsc::unbounded_channel();
+
+        let con: DynBackCon = Arc::new(Self { pub_key, send });
+
+        let con_recv: DynBackConRecvData = Box::new(ConRecvData(recv));
+
+        (con, con_recv)
+    }
+}
+
+impl BackCon for Con {
+    fn send(&self, data: Vec<u8>) -> BoxFuture<'_, Result<()>> {
+        let r = self.send.send(data).map_err(|err| {
+            std::io::Error::other(format!("send failure ({err:?})"))
+        });
+        Box::pin(async { r })
+    }
+
+    fn pub_key(&self) -> &PubKey {
+        &self.pub_key
+    }
+
+    fn is_using_webrtc(&self) -> bool {
+        true
+    }
+
+    fn get_stats(&self) -> tx5_connection::ConnStats {
+        tx5_connection::ConnStats {
+            send_msg_count: 0,
+            send_byte_count: 0,
+            recv_msg_count: 0,
+            recv_byte_count: 0,
+        }
+    }
+}
+
+struct WaitCon {
+    pub_key: PubKey,
+    con: Option<DynBackCon>,
+    con_recv: Option<DynBackConRecvData>,
+}
+
+impl BackWaitCon for WaitCon {
+    fn wait(
+        &mut self,
+        _recv_limit: Arc<tokio::sync::Semaphore>,
+    ) -> BoxFuture<'static, Result<(DynBackCon, DynBackConRecvData)>> {
+        let con = self.con.take().unwrap();
+        let con_recv = self.con_recv.take().unwrap();
+        Box::pin(async { Ok((con, con_recv)) })
+    }
+
+    fn pub_key(&self) -> &PubKey {
+        &self.pub_key
+    }
+}
+
+struct EpRecvCon(tokio::sync::mpsc::UnboundedReceiver<DynBackWaitCon>);
+
+impl BackEpRecvCon for EpRecvCon {
+    fn recv(&mut self) -> BoxFuture<'_, Option<DynBackWaitCon>> {
+        Box::pin(async { self.0.recv().await })
+    }
+}
+
+fn gen_pub_key(uniq: u64, loc: usize) -> PubKey {
+    // in base64 a string of 0xb7 renders as `t7e3t7e3t...`
+    // which sort-of looks like "test" if you squint : )
+    let mut pub_key = [0xb7; 32];
+    pub_key[16..24].copy_from_slice(&uniq.to_be_bytes());
+    pub_key[24..32].copy_from_slice(&(loc as u64).to_be_bytes());
+
+    PubKey(Arc::new(pub_key))
+}
+
+fn parse_pub_key(pub_key: &PubKey) -> (u64, usize) {
+    let mut uniq = [0; 8];
+    uniq.copy_from_slice(&(*pub_key)[16..24]);
```

Comment:
Is the `*` dereference necessary? Can't we just do:

```suggestion
    uniq.copy_from_slice(&pub_key[16..24]);
```